js/main.js

```diff
@@ -91,6 +92,10 @@ function setupCollectionControls() {
     localStorage.setItem("page-size", CARDS_PER_PAGE_DROPDOWN.value);
     toggleCollection();
   };
+
+  SEARCH_BAR.oninput = (event) =>
+    showCollection(cards_data, COLLECTIONS_MAIN_CONTENT, 1, event.target.value);
+
```

Comment:
if it's not too much extra work, maybe add a debounce here?

Reply:
Changed it from "oninput" to "onchange". oninput executes on every character while onchange executes when pressing enter or losing focus, so it executes only once per query without having to account for edge cases while debouncing.